artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/PagedReferenceImpl.java

```diff
@@ -120,6 +128,8 @@ public PagedReferenceImpl(final PagePosition position,
          this.largeMessage = message.getMessage().isLargeMessage() ? IS_LARGE_MESSAGE : IS_NOT_LARGE_MESSAGE;
          this.transactionID = message.getTransactionID();
          this.messageID = message.getMessage().getMessageID();
+         this.priority = message.getMessage().getPriority();
```

Comment:
deliveryTime can be set in the constructor like transactionID , messageID , etc :)